stable/pomerium/templates/tls-secrets.yaml

```diff
@@ -16,3 +16,129 @@ data:
 ---
 {{- end }}
 {{- end }}
+
+{{- define "pomerium.authenticate.tlsSecretObject" -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ template "pomerium.name" . }}
+    helm.sh/chart: {{ template "pomerium.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+  annotations:
+{{- if .Values.config.generateTLS }}
+    helm.sh/hook: pre-install
+{{- end }}
+  name: {{ template "pomerium.authenticate.tlsSecret.name" . }}
+  namespace: {{ .Release.Namespace }}
+type: kubernetes.io/tls
+data:
+{{- end -}}
+
+{{- define "pomerium.authorize.tlsSecretObject" -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ template "pomerium.name" . }}
+    helm.sh/chart: {{ template "pomerium.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+  annotations:
+{{- if .Values.config.generateTLS }}
+    helm.sh/hook: pre-install
+{{- end }}
+  name: {{ template "pomerium.authorize.tlsSecret.name" . }}
+  namespace: {{ .Release.Namespace }}
+type: kubernetes.io/tls
+data:
+{{- end -}}
+
+{{- define "pomerium.proxy.tlsSecretObject" -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ template "pomerium.name" . }}
+    helm.sh/chart: {{ template "pomerium.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+  annotations:
+{{- if .Values.config.generateTLS }}
+    helm.sh/hook: pre-install
+{{- end }}
+  name: {{ template "pomerium.proxy.tlsSecret.name" . }}
+  namespace: {{ .Release.Namespace }}
+type: kubernetes.io/tls
+data:
+{{- end -}}
+
+{{- define "pomerium.ca.tlsSecretObject" -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ template "pomerium.name" . }}
+    helm.sh/chart: {{ template "pomerium.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+  annotations:
+{{- if .Values.config.generateTLS }}
+    helm.sh/hook: pre-install
+{{- end }}
+  name: {{ template "pomerium.caSecret.name" . }}
+  namespace: {{ .Release.Namespace }}
+type: Opaque
+data:
+{{- end -}}
+
+{{- if not .Values.existingLegacyTLSSecret }}
+{{- if (and .Values.config.generateTLS .Release.IsInstall) }}
```

Comment:
@travisgroth one last question: will the `.Release.IsInstall` be true if this is actually an upgrade from a previous version of the chart?

Reply:
Ah it would be false.  We will need a flag to force it to prevent requiring reinstall.  Easy to do.

Reply:
This exposed a little complexity in the generated upgrade procedure.  I think it is addressed now.  The upgrade and tls cert docs should cover it - see if they make sense.